src/main/java/com/cms/example/cms/feature/user/UserService.java

```diff
@@ -0,0 +1,177 @@
+import com.cms.example.cms.feature.geo.UpazilaRepository;
+import com.cms.example.cms.feature.subject.SubjectRepository;
+import com.cms.example.cms.feature.userRating.UserRatingRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final UserRatingRepository userRatingRepository;
+    private final DivisionRepository divisionRepository;
+    private final DistrictRepository districtRepository;
+    private final UpazilaRepository upazilaRepository;
+    private final SubjectRepository subjectRepository;
+    private final AcademicInfoRepository academicInfoRepository;
+
+    @Transactional
+    public CmsUser saveCmsUser(CmsUser cmsUser) {
+        if (UserRating.isNonNull(cmsUser.getUserRating())) {
+            cmsUser.setUserRating(userRatingRepository.getOne(cmsUser.getUserRating().getUserRatingId()));
+        }
+        populateAddress(cmsUser);
+        populateAcademicInfo(cmsUser);
+        cmsUser = userRepository.save(cmsUser);
+        return getCmsUserById(cmsUser.getCmsUserId());
+    }
+
+    private void populateAddress(CmsUser cmsUser) {
+        if (CollectionUtils.isEmpty(cmsUser.getAddresses())) return;
+        cmsUser.getAddresses().forEach(address -> {
+            if (Division.isNonNull(address.getDivision())) {
+                address.setDivision(divisionRepository.getOne(address.getDivision().getDivisionId()));
+            }
+
+            if (District.isNonNull(address.getDistrict())) {
+                address.setDistrict(districtRepository.getOne(address.getDistrict().getDistrictId()));
+            }
+
+            if (Upazila.isNonNull(address.getUpazila())) {
+                address.setUpazila(upazilaRepository.getOne(address.getUpazila().getUpazilaId()));
+            }
+            address.setCmsUser(cmsUser);
+        });
+    }
+
+    private void populateAcademicInfo(CmsUser cmsUser) {
+        if (CollectionUtils.isEmpty(cmsUser.getAcademicInfos())) return;
+        cmsUser.getAcademicInfos().forEach(academicInfo -> {
+            populateSubject(academicInfo);
+            academicInfo.setCmsUser(cmsUser);
+        });
+    }
+
+    private void populateSubject(AcademicInfo academicInfo) {
+        List<Subject> subjectList = new ArrayList<>();
+        academicInfo.getSubjects().forEach(subject -> {
+            if (Subject.isNonNull(subject)) {
+                Subject retrievedSubject = subjectRepository.getOne(subject.getSubjectId());
+                subjectList.add(retrievedSubject);
+            }
+        });
+        academicInfo.setSubjects(subjectList);
```

Comment:
first clear the list then set

```suggestion
        academicInfo.getSubjects().clear();
        academicInfo.setSubjects(subjectList);
```

Reply:
updated